packages/hop-node/src/watchers/classes/Bridge.ts

```diff
@@ -648,22 +672,29 @@ export default class Bridge extends ContractBase {
   }
 
   private readonly getBlockValues = async (options: Partial<EventsBatchOptions>, state?: State): Promise<BlockValues> => {
-    const { startBlockNumber, endBlockNumber } = options
+    const { startBlockNumber, endBlockNumber, syncCacheKey } = options
 
     let end: number
     let start: number
     let totalBlocksInBatch: number
     const { totalBlocks, batchBlocks } = globalConfig.sync[this.chainSlug]
-    let blockNumberWithAcceptableFinality: number
 
-    if (getHasFinalizationBlockTag(this.chainSlug)) {
+    // TODO: Better state handling
+    const isHeadSync = syncCacheKey?.includes(HeadSyncKeySuffix)
```

Comment:
suggestion: use `startsWith` instead of `includes`

```suggestion
const isHeadSync = syncCacheKey?.startsWith(HeadSyncKeySuffix)
```

optional: a step further would be to move to separate function

```ts
const isHeadSync = this.isHeadSyncCacheKey(syncCacheKey)
```

Reply:
Going to keep for now with the thought that the next update to this logic should be a bigger update that will have more data about how the head sync is used in practice and will likely include ripping that out into its own function. (I don't love that it is just a suffix now, but it definitely works).

Reply:
Resolved with 2c22d36cd66155375efc707a3572fe8fda6fc1fc